### ISSUE TITLE
Again in Ch 5, 6, 7

### ISSUE THREAD

**Issue by deulkkae** [gedeck/practical-statistics-for-data-scientists#16]
### Naive Bayes, The Naive Solution 
The predicted probabilities results are different. They should be 0.4798964(paid off) 0.5201036(default). 

I ran the code in colab. Would check this notebook?
https://colab.research.google.com/drive/1ChitMlzaMHYDru6ngI1qBHhJGcIP-RhI#scrollTo=1EnynWD14l2R&line=7&uniqifier=1

### Variable importance
Need line-break in line 318. 
https://github.com/gedeck/practical-statistics-for-data-scientists/blob/3e1bf1cb3ca3f345195f702addc3a3b01e67c58c/python/code/Chapter%206%20-%20Statistical%20Machine%20Learning.py#L318

### Hyperparameters and Cross-Validation
Need line-break in 453.
https://github.com/gedeck/practical-statistics-for-data-scientists/blob/3e1bf1cb3ca3f345195f702addc3a3b01e67c58c/python/code/Chapter%206%20-%20Statistical%20Machine%20Learning.py#L453

And line 452 has type error. Would check this line?
"TypeError: Object with dtype category cannot perform the numpy op subtract"

### Python XGBoost codes in Ch6
It would be better to set eval_metric='error' in Python codes too.

**Comment by gedeck** [comment]
**Naive Bayes:**
I checked this and I also get these changed values with a fresh Installation of R. As the results were practically identical between the Python and R implementations, I reached out the author of the klaR package to find out if something has changed that would explain this large change. The changelog of the klaR package mentions changes to the Laplace smoothing that may be responsible to this. When I set alpha in Python to about 2000, I get similar results to R now. I wait what the author of the R package responds.

**Formatting changes**
I'm not going to change the formatting. The main reason is that the same code is also used in the book and I try to keep it as concise/short as possible.

**XGBoost / type error line 452**
I thought I already added eval_metric='error' to the notebook. Have done it in another notebook but not here. I've done it now. At the same time, I addressed the warning message that the label encoder will be deprecated in xgboost and therefore changed the definition of `y` to:
```
X = pd.get_dummies(loan_data[predictors], drop_first=True)
y = pd.Series([1 if o == 'default' else 0 for o in loan_data[outcome]])
```
That seemed to have also resolved the type error, as I don't see it.

**Comment by gedeck** [comment]
**Naive Bayes (update)**:
I fortunately had an older installation of R around, so I could confirm that with an older version, the results are as printed in the book. The old version was R 3.6.1 and new version R 4.0.3. One thing that changed between these two versions is the handling of string columns when reading data files. The difference is in `default.stringsAsFactors()` which is TRUE in 3.6.1 and FALSE in 4.0.3. When I explicitly set 
```
loan_data <- read.csv(file.path(PSDS_PATH, 'data', 'loan_data.csv.gz'), stringsAsFactors=TRUE)
```
I get the original results printed in the book. 
```
$posterior
      paid off   default
[1,] 0.3463013 0.6536987
```

**Comment by gedeck** [comment]
Closed with 7b89f15a73786d11c05bb13d3635821215ffc34a
